src/azure-cli/azure/cli/command_modules/hdinsight/custom.py

```diff
@@ -68,7 +68,7 @@ def create_cluster(cmd, client, cluster_name, resource_group_name, cluster_type,
 
     if component_version:
         # See validator
-        component_version = {c: v for c, v in [version.split('=') for version in component_version]}
+        component_version = dict([version.split('=') for version in component_version])
```

Comment:
The constructor of `dict` takes `iterables` of 2-element `iternables`.

https://docs.python.org/3/library/stdtypes.html#dict